spec/controllers/instructeurs/procedures_controller_spec.rb

```diff
@@ -1122,4 +1122,210 @@
       expect(response.body).to include("lundi 17 février à 10h00")
     end
   end
+
+  describe '#history' do
+    let(:instructeur) { create(:instructeur) }
+    let(:procedure) { create(:procedure, :published, instructeurs: [instructeur]) }
+
+    before do
+      sign_in(instructeur.user)
+      procedure.revisions.update_all(published_at: nil)
+
+      unless InstructeursProcedure.exists?(instructeur: instructeur, procedure: procedure)
+        create(:instructeurs_procedure, instructeur: instructeur, procedure: procedure)
+      end
+    end
+
+    context 'when there are no published revisions' do
+      before do
+        get :history, params: { procedure_id: procedure.id }
+      end
+
+      it 'assigns an empty comparison_pairs array' do
+        expect(assigns(:comparison_pairs)).to eq([])
+      end
+    end
+
+    context 'when there is only one published revision' do
+      before do
+        create(:procedure_revision, procedure: procedure, published_at: 1.day.ago)
+        get :history, params: { procedure_id: procedure.id }
+      end
+
+      it 'assigns an empty comparison_pairs array' do
+        expect(assigns(:comparison_pairs)).to eq([])
+      end
+    end
+
+    context 'when there are two published revisions' do
+      let!(:old_revision) { create(:procedure_revision, procedure: procedure, published_at: 2.days.ago) }
+      let!(:new_revision) { create(:procedure_revision, procedure: procedure, published_at: 1.day.ago) }
+
+      before do
+        get :history, params: { procedure_id: procedure.id }
+      end
+
+      it 'assigns a comparison_pairs array with one pair' do
+        expect(assigns(:comparison_pairs).length).to eq(1)
+      end
+
+      it 'includes the correct revisions in the pair' do
+        pair = assigns(:comparison_pairs).first
+        expect(pair).to eq([new_revision, old_revision])
+      end
+    end
+
+    context 'when there are multiple published revisions' do
+      let!(:oldest_revision) { create(:procedure_revision, procedure: procedure, published_at: 4.days.ago) }
+      let!(:middle_revision) { create(:procedure_revision, procedure: procedure, published_at: 3.days.ago) }
+      let!(:recent_revision) { create(:procedure_revision, procedure: procedure, published_at: 2.days.ago) }
+      let!(:newest_revision) { create(:procedure_revision, procedure: procedure, published_at: 1.day.ago) }
+
+      before do
+        get :history, params: { procedure_id: procedure.id }
+      end
+
+      it 'assigns a comparison_pairs array with the correct number of pairs' do
+        expect(assigns(:comparison_pairs).length).to eq(3)
```

Comment:
ici juste tester qu'on a ttes revisions devrait aller 
```suggestion
        expect(assigns(:comparison_pairs).length).to eq(3)
```